am-lib/src/main/java/uk/gov/hmcts/reform/amlib/internal/PermissionsService.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.amlib.internal;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import uk.gov.hmcts.reform.amlib.enums.Permission;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PermissionsService {
+
+    public Map<JsonPointer, Set<Permission>> mergePermissions(List<Map<JsonPointer, Set<Permission>>> records) {
+
+        Map<JsonPointer, Set<Permission>> map = new ConcurrentHashMap<>();
```

Comment:
Below solution seems quite attractive:

```
Map<String, Employee> map3 = Stream.of(map1, map2)
  .flatMap(map -> map.entrySet().stream())
  .collect(Collectors.toMap(
    Map.Entry::getKey,
    Map.Entry::getValue,
    (v1, v2) -> new Employee(v1.getId(), v2.getName())));
```

Reply:
I did try an implementation like this at first and then replaced it with the current solution